lib/elixir/lib/module.ex

```diff
@@ -673,7 +672,7 @@ defmodule Module do
 
   defp simplify_signature({var, _, atom}, acc) when is_atom(atom) do
     case Atom.to_string(var) do
-      "_" <> rest -> {{String.to_atom(rest), [], Elixir}, acc}
```

Comment:
The third element of this tuple is the variable context. It doesn't matter in the scenario of pretty printing a variable. We are using it to specify the variable priority (Elixir for low priority, nil for high). See here: https://github.com/eksperimental/elixir/blob/7c4376ee6f2afd6bc76bfcc11e103bc1a2c10af7/lib/elixir/lib/module.ex#L764-L772

So the fix is likely in merge_signatures. You can see the last clause which is where we generate the `arg` one.